### PR TITLE
Issue 24 lz4 frame module

### PR DESCRIPTION
**Pull request by tzulberti-jampp**
Solve #24

### CODE REVIEW

**Review by klaussfreire — CHANGES_REQUESTED** [review]
Nope. They're not interchangeable. A change like this needs far more care and at least a minor bump. Also a version change in the file format, and compatibility code when reading the old format.

**Comment by tzulberti-jampp** [comment]
If I understand correctly the version bump is required because what the file created with lz4 0.8.1 is going to be imcompatible with the one created with version 0.18.1. Not because of the difference on code...

I googled who to make lz4 compatible between versions but I wasn't able to find any information about this. Do you have an idea on who to make those versions compatible?

**Comment by klaussfreire** [comment]
Yes. Use the file format version embedded in the bundles and have both versions of the code.

**Comment by klaussfreire** [comment]
It will probably be more efficient to talk about it in person tomorrow

File: sharedbuffers/mapped_struct.py [hunk 1]
```diff
@@ -298,8 +297,15 @@ def unpack_from(cls, buf, offs, idmap = None, array = array.array, itemsizes = {
             raise ValueError("Inconsistent data, unknown type code %r" % (dcode,))
         return rv
 
-lz4_decompress = cython.declare(object, lz4.decompress)
-lz4_compress = cython.declare(object, lz4.compress)
+try:
+    import lz4.block
+    lz4_functions_module = lz4.block
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py]: lz4_block?

**Review by klaussfreire — CHANGES_REQUESTED** [review]

**Review by klaussfreire — APPROVED** [review]

**Review by zeitos — APPROVED** [review]
LGTM

**Pull request merged**